src/main/java/com/ericsson/bss/cassandra/ecaudit/auth/AuditWhitelistManager.java

```diff
@@ -83,16 +76,14 @@ public void createRoleWhitelist(AuthenticatedUser performer, RoleResource role,
             Map<String, Set<String>> addStatements = new HashMap<>();
             for (Map.Entry<String, String> optionEntry : options.getCustomOptions().get().entrySet())
             {
-                if (!VALID_CREATE_OPTIONS.containsKey(optionEntry.getKey()))
-                {
-                    throw new InvalidRequestException("Invalid create user option: " + optionEntry.getKey());
-                }
-
-                Set<IResource> resources = tryConvertToResourceSet(optionEntry.getValue());
+                WhitelistOperation whitelistOperation = whitelistOptionParser.parseWhitelistOperation(optionEntry.getKey());
+                Set<IResource> resources = whitelistOptionParser.parseResource(optionEntry.getValue());
+                String operation = whitelistOptionParser.parseTargetOperation(optionEntry.getKey());
 
+                whitelistContract.verifyCreateRoleOption(whitelistOperation);
```

Comment:
There is actually no test case covering this. There is one that looks like it's supposed to (testRevokeAtCreateIsDenied) but it is failing due to "Invalid whitelist option: revoke_audit_whitelist_for_select" rather than the revoke option.